_posts/2024-04-15-video-segmentation.md

```diff
@@ -0,0 +1,176 @@
+
+As usually I will be trying to update this post with the most recent papers and code implementations I find interesting.
+
+## Introduction to Video Segmentation
+
+Video segmentation condenses a lot of different tasks which can have multiple names. I personaly like the taxonomy used in the [Youtube-VOS Dataset](https://youtube-vos.org/dataset/) which is one of the main benchmarks in this field so I will stick with it through this post. The different tasks are:
+
+- **Video Object Segmentation (VOS)**: targets at segmenting a particular object instance throughout the entire video sequence given only the object mask of the first frame.
+- **Video Instance Segmentation (VIS)**: extends image instance segmentation to videos, aiming to segment and track object instances across frames.
+- **Referring Video Object Segmentation (RVOS)**: is a task that requires to segment a particular object instance in a video given a natural language expression that refers to the object instance.
+
+<div class="post-center-image">
+<a href="/assets/svg/posts/2024-04-15-video-segmentation/video-segmentation-types-diagram.svg" class="lightbox-image" title="Diagram with the different video segmentation methods">
+ <img src="/assets/svg/posts/2024-04-15-video-segmentation/video-segmentation-types-diagram.svg" loading="lazy" alt="Diagram with the different video segmentation methods">
+</a>
+</div>
+
+{:refdef: class="image-caption"}
+*Diagram with the different video segmentation methods*
+{: refdef}
+
+
+## Key concepts
+
+### Input sequence length
+
+The input sequence length is crucial in video segmentation. Longer sequences provide more context for accurately segmenting objects, even during occlusion and appearance changes. However, they require more computational power and increase training and inference times. Think that transformers have a quadratic complexity with respect to the sequence length so most of the models we are going to discuss are trained with very short clips (mostly between 2 and 8 frames).
+
+### Stride
+
+The input sequence length defines the number of frames processed in parallel. The stride is the one that determines the temporal distance between frames in the input sequence. A stride of 1 means that the input sequence is a continuous sequence of frames, while a stride of 2 means that every other frame is skipped. By increasing the stride the system can work faster because it's looking at fewer frames.
+
+### Offline vs Online
+
+Many video segmentation approaches are categorized as offline, analyzing the entire video in one go, ideal for short videos and dependent on the model's capacity for the longest clip. On the other hand online methods divide videos into overlapping chunks, and results from these segments are merged using a rule-based post-tracking method. This approach ensures continuous tracking across the video by analyzing and linking instances from overlapping segments.
```

Comment:
I don't understand what you mean by **_`and dependent on the model's capacity for the longest clip`_** the bold sentence. You mean they're limited by the maximum length the model can process?

```suggestion
Many video segmentation approaches are categorized as offline. They process the entire video in one go, which is ideal for short videos and dependent on the model's capacity for the longest clip. On the other hand online methods divide videos into overlapping chunks, and results from these segments are merged using a rule-based post-tracking method. This approach ensures continuous tracking across the video by processing and linking instances from overlapping segments.
```

Reply:
Yup. But do not really like it either, I will rephrase this